spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/export/graphite/GraphiteMetricsExportAutoConfiguration.java

```diff
@@ -58,9 +60,15 @@ public GraphiteConfig graphiteConfig(GraphiteProperties graphiteProperties) {
 
 	@Bean
 	@ConditionalOnMissingBean
-	public GraphiteMeterRegistry graphiteMeterRegistry(GraphiteConfig graphiteConfig,
-			Clock clock) {
-		return new GraphiteMeterRegistry(graphiteConfig, clock);
+	public HierarchicalNameMapper graphiteHierarchicalNameMapper(GraphiteConfig config) {
+		return new GraphiteHierarchicalNameMapper(config.tagsAsPrefix());
+	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	public GraphiteMeterRegistry graphiteMeterRegistry(GraphiteConfig config, Clock clock,
+			HierarchicalNameMapper nameMapper) {
+		return new GraphiteMeterRegistry(config, clock, nameMapper);
```

Comment:
If we're going to make this change for Graphite, should it not also be made for all the other meter registries that take a `HierarchicalNameMapper`?

Reply:
@wilkinsona It sounds right to me.

@jkschneider WDYT?

Reply:
I believe all other hierarchical implementations don't have a specialized type `GraphiteHierarchicalNameMapper`. At one point, we had this in the configurations but removed it because injecting one name mapper affects them all and it was easy enough to wire your own registry including a custom name mapper.

Reply:
Thanks, @jkschneider. So we could type the injection into the `@Bean` method more strongly by using `GraphiteHierarchicalNameMapper` or we could just not do anything and let people configure their own registry. Either way, you're declaring a single bean (your own name mapper or your own registry). I'm tempted to say that it's not worth making a change here and recommending that people just configure their own registry.

Reply:
I'd prefer to keep things consistent across registries rather than doing something that's Graphite-specific. A change needs to be made in the Spring Boot legacy support regardless as it's currently too weakly typed.

Reply:
Removed in the legacy module for consistency in https://github.com/micrometer-metrics/micrometer/commit/659ac962cdb920d0c1f50533075360c1b2646765, which will be included in micrometer-spring-legacy 1.2.0

Reply:
@wilkinsona @jkschneider Thanks for the feedback! Closing as discussed.